### PR TITLE
Sauce Visual API lifecycle

### PR DESCRIPTION
**Pull request by kb-kerem**
### Description
To help customers that don’t rely on a technology we provide a binding for, we provide a (basic) documentation of Sauce Visual API lifecycle.

### Types of Changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation fix (typos, incorrect content, missing content, etc.)

### CODE REVIEW

File: docs/visual-testing/workflows/api-lifecycle.md [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+      "id": "snapshot-id-here",
+      "uploadId": "upload-id-here"
+    }
+  }
+}
+```
+
+- `id`: The ID of the snapshot that has been created.
+
+### 4. Finish Build
```
Comment (konraddysput) [inline docs/visual-testing/workflows/api-lifecycle.md:135]: - can we add information we can close the build automatically after a few hours?
- can we point out this is necessary step and the build cannot be re-used after closing it?

**Review by konraddysput — APPROVED** [review]

File: docs/visual-testing/workflows/api-lifecycle.md [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+```
+
+- `status`: As a newly created build without any snapshots, this will be `RUNNING`.
+- `url`: The URL that can be used any time to check the build on Sauce Labs.
+
+### 2. Upload Image
+
+Next, upload an image to the build. This is a two step process. 
+
+First, get a placeholder URL for image upload using `createSnapshotUpload`.
```
Comment (FriggaHel) [inline docs/visual-testing/workflows/api-lifecycle.md:58]:
```suggestion
First, obtain a signed URL for uploading your image by using the `createSnapshotUpload` mutation.
```

File: docs/visual-testing/workflows/api-lifecycle.md [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+
+**GraphQL Mutation:**
+
+```graphql
+mutation {
+  createSnapshotUpload(input: {buildId: "build-id-here"}) {
+    id
+    uploadUrl
```
Comment (FriggaHel) [inline docs/visual-testing/workflows/api-lifecycle.md:66]: Shouldn't we mention `domUploadUrl` too ?

File: docs/visual-testing/workflows/api-lifecycle.md [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+      "uploadUrl": "upload-url-here"
+    }
+  }
+}
+```
+
+- `id`: Upload ID to use in the subsequent steps.
+- `uploadUrl`: The URL to upload the image in the next step.
+
+Next, send a `PUT` request to `uploadUrl` with image file in the body of the request. Only **PNG** files are supported.
```
Comment (FriggaHel) [inline docs/visual-testing/workflows/api-lifecycle.md:89]: Should we mention the support for `Content-MD5` header ?

It's useful when it comes to ensuring the image has been correctly uploaded

File: docs/visual-testing/workflows/api-lifecycle.md [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+
+**cURL Representation:**
+
+```sh
+curl --request PUT \
+  --url 'upload-url-here' \
+  --header 'Content-Type: image/png' \
+  --data 'png-file-content'
```
Comment (FriggaHel) [inline docs/visual-testing/workflows/api-lifecycle.md:97]: Alternative representation:

```suggestion
  --data '@my-screenshot.png'
```

(Starting with `@` will read the file).

File: docs/visual-testing/workflows/api-lifecycle.md [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+```
+
+### 3. Create Snapshot
+
+After uploading an image, create a snapshot to capture the current state of the build.
```
Comment (FriggaHel) [inline docs/visual-testing/workflows/api-lifecycle.md:102]: I find that part not super clear. What actually means "to capture the current state of the build" ?

Suggestion (may need improvements):
```suggestion
After uploading your image, add your snapshot and its metadata to your build.
```

File: docs/visual-testing/workflows/api-lifecycle.md [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+
+**GraphQL Mutation:**
+
+```graphql
+mutation {
+  createSnapshot(
+    input: {buildUuid: "build-id-here", uploadId: "upload-id-here", name: "Your snapshot name"}
```
Comment (FriggaHel) [inline docs/visual-testing/workflows/api-lifecycle.md:109]: We should mention at least `branch`, `project` fields. That's dimensions that we expect users to use.

We should also strongly advise to fill in `OS` / `Browser`.

File: docs/visual-testing/workflows/api-lifecycle.md [hunk 8]
```diff
@@ -0,0 +1,167 @@
...
+    }
+  }
+}
+```
+
+- `id`: The ID of the snapshot that has been created.
+
+### 4. Finish Build
+
+Finally, finish the build process to mark it as complete.
```
Comment (FriggaHel) [inline docs/visual-testing/workflows/api-lifecycle.md:137]: Probably better this way:
```suggestion
Finally, finish the build to mark it as complete.
```

File: docs/visual-testing/workflows/api-lifecycle.md (hunk 7 above)
Comment (kb-kerem) [reply docs/visual-testing/workflows/api-lifecycle.md:109]: `branch` and `project` are build specific properties, not snapshot. I'll mention them there.

**Comment by kb-kerem** [comment]
@konraddysput @FriggaHel I've addressed all the points you've raised. Would appreciate another review.

**Review by omacranger — APPROVED** [review]
Looks great to me.

**Review by FriggaHel — APPROVED** [review]

**Pull request merged**